### ISSUE TITLE
'factory.favicon' not work

### ISSUE THREAD

**Issue by paulzzh** [barneygale/quarry#48]
My code:
(It's a copy of 'example/server_downtime.py')

from twisted.internet import reactor
from quarry.net.server import ServerFactory, ServerProtocol
class DowntimeProtocol(ServerProtocol):
    def packet_login_start(self, buff):
        buff.discard()
        #self.close(self.factory.motd)
        self.close("balabalbalabalabala...")
        logger.info("someone tryed to join,Start Server")
        reactor.callLater(3,reactor.stop)
class DowntimeFactory(ServerFactory):
    protocol = DowntimeProtocol

def waitmain():
    # Create factory
    factory = DowntimeFactory()
    factory.motd = "balabalabala...."
    factory.max_players = 8
    factory.online_mode = False
    factory.favicon = 'icon.png'
    # Listen
    factory.listen("0.0.0.0", 25565)
    reactor.run()

waitmain()

In fact, it doesn't work very well:
![image](https://user-images.githubusercontent.com/28055560/43214805-f58e5846-906c-11e8-9e6e-b9bcbaf7315d.png)
I have delete the 'server.dat' ,but it‘s useless..
I 'm sure I have put a 'icon.png' with the script.
This 'icon.png' works well in a 1.13 vanilla server.

**Comment by wgaylord** [comment]
You might have to actually do the base64 encoding if the image in the field for it to work.

**Comment by dries007** [comment]
This is due to a MC bug, the protocol has remained but something about the data has changed.
The relevant issue on the MC tracker has been reopened: https://bugs.mojang.com/browse/MC-121282
After inspection with wireshark I found that if you copy-paste the data the vanilla client sends, it works correctly, but that data is not a plain 1:1 base64 encoding of the PNG anymore.

Another note about the relevant code: It's not really efficient to reload the file every launch, so I'm locally patching the code to do this now, which also allows me to paste in a working string and get an icon for the time being:

```python
# in file 'quarry/net/server.py' at line 255:
        if self.factory.favicon is not None:
            d["favicon"] = self.factory.favicon # todo: Patch! Also, stop using 'base64.encodestring', it's deprecated and adds \n's for no reason.
            #with open(self.factory.favicon, "rb") as fd:
            #    d["favicon"] = "data:image/png;base64," + base64.encodestring(
            #        fd.read()).decode('ascii')
```

**Comment by barneygale** [comment]
Fixed in #49, thanks @dries007!
